WalletWasabi.Packager/MacSignTools.cs

```diff
@@ -0,0 +1,394 @@
+			var appName = "Wasabi Wallet.app";
+			var appPath = Path.Combine(dmgPath, appName);
+			var binPath = Path.Combine(appPath, "Contents", "MacOS");
+			var resPath = Path.Combine(appPath, "Contents", "Resources");
+			var infoFilePath = Path.Combine(appPath, "Contents", "Info.plist");
+			var dmgFileName = $"Wasabi-{versionPrefix}.dmg";
+			var dmgFilePath = Path.Combine(workingDir, dmgFileName);
+			var appNotarizeFilePath = Path.Combine(workingDir, $"Wasabi-{versionPrefix}.zip");
+			var contentsPath = Path.GetFullPath(Path.Combine(Program.PackagerProjectDirectory.Replace("\\", "//"), "Content", "Osx"));
+			var entitlementsPath = Path.Combine(contentsPath, "entitlements.plist");
+			var entitlementsSandBoxPath = Path.Combine(contentsPath, "entitlements_sandbox.plist");
+			var torZipDirPath = Path.Combine(binPath, "TorDaemons", "tor-osx64");
+			var torZipPath = $"{torZipDirPath}.zip";
+			var desktopDmgFilePath = Path.Combine(desktopPath, dmgFileName);
+
+			var signArguments = $"--sign \"L233B2JQ68\" --verbose --options runtime --timestamp";
+
+			Console.WriteLine("Phase: creating the working directory.");
+
+			Console.WriteLine("Enter appleId (email):");
+			var appleId = Console.ReadLine();
+			Console.WriteLine("Enter password:");
+			var password = Console.ReadLine();
+
+			if (Directory.Exists(workingDir))
+			{
+				DeleteWithChmod(workingDir);
+			}
+
+			if (File.Exists(desktopDmgFilePath))
+			{
+				File.Delete(desktopDmgFilePath);
+			}
+
+			Console.WriteLine("Phase: creating the app.");
+
+			IoHelpers.EnsureDirectoryExists(binPath);
+			ZipFile.ExtractToDirectory(zipPath, binPath); // Copy the binaries.
+
+			IoHelpers.CopyFilesRecursively(new DirectoryInfo(Path.Combine(contentsPath, "App")), new DirectoryInfo(appPath));
+
+			Console.WriteLine("Update the plist file with current information for example with version.");
+
+			var lines = File.ReadAllLines(infoFilePath);
+			string bundleIdentifier = null;
+
+			for (int i = 0; i < lines.Length; i++)
+			{
+				string line = lines[i];
+				if (!line.TrimStart().StartsWith("<key>", StringComparison.InvariantCultureIgnoreCase))
+				{
+					continue;
+				}
+
+				if (line.Contains("CFBundleShortVersionString", StringComparison.InvariantCulture) ||
+					line.Contains("CFBundleVersion", StringComparison.InvariantCulture))
+				{
+					lines[i + 1] = lines[i + 1].Replace("?", $"{Version.Parse(versionPrefix).ToString(3)}"); // Apple allow only 3 version tags in plist.
+				}
+				else if (line.Contains("CFBundleIdentifier", StringComparison.InvariantCulture))
+				{
+					bundleIdentifier = lines[i + 1].Trim().Replace("<string>", "").Replace("</string>", "");
+				}
+			}
+			IoHelpers.DeleteRecursivelyWithMagicDustAsync(infoFilePath).GetAwaiter().GetResult();
+			File.WriteAllLines(infoFilePath, lines);
+
+			using (var process = Process.Start(new ProcessStartInfo
+			{
+				FileName = "chmod",
+				Arguments = $"-R u+rwX,go+rX,go-w \"{appPath}\"",
```

Comment:
This is good and we should use this syntax in other places too instead of the numeric notation (`chmod 756`)

Reply:
Agree. By the way, what is the capital X there?

Reply:
This `X` is used when you want to apply the `enter` permission to directories recursively without affecting the permission of the files contained in those directories. If you use lowercase `x` recursively then you change absolutely every `execute` permission to all the files and directories (a disaster!).